### DISCUSSION TITLE
Is mid-circuit measurement available in tket/pytket?

### DISCUSSION THREAD

**Discussion by jagandecapri** [CQCL/tket#957] [Q&A] [answered]
Hi, mid-circuit measurement as the paper [Qubit-reuse compilation with mid-circuit measurement and reset](https://arxiv.org/abs/2210.08039) is available in tket/pytket?

**Comment by cqc-alec** [comment, marked as answer]
Yes, mid-circuit measurement and reset is fully supported by pytket. Whether a specific backend supports it is another question. Some do, such as the Quantinuum H-series devices targeted by the [pytket-quantinuum](https://github.com/CQCL/pytket-quantinuum) backend (as used in that paper).

**Reply by jagandecapri** [reply]
Hi @cqc-alec , thanks for the info. Is the optimization algorithm proposed in that paper is also implemented in tket/pytket?

**Reply by cqc-alec** [reply]
At the moment the optimization algorithm is implemented in TKET as a compiler pass available only to H-series users on the Quantinuum user portal; it has not been open-sourced.

**Reply by jagandecapri** [reply]
Thank you, @cqc-alec, for the information. 💯
